components/brave_wallet/common/eth_sign_typed_data_helper.cc

```diff
@@ -278,7 +279,7 @@ EthSignTypedDataHelper::EncodeField(const std::string_view type,
     }
     std::vector<uint8_t> bytes;
     CHECK(PrefixedHexStringToBytes(*value_str, &bytes));
-    if (bytes.size() > 32) {
+    if (bytes.size() != num_bytes) {
```

Comment:
Actually per spec https://eips.ethereum.org/EIPS/eip-712#definition-of-encodedata
that should be exactly 32 bytes. 
So we should pick first num_bytes and check that trailing part is all zero